### ISSUE TITLE
Ошибка "Out of memory" при конвертации Московской базы

### ISSUE THREAD

**Issue by lagytenko** [mbry/DgdatToXlsx#7]
Добрый день!
Большое спасибо за конвертер, очень полезный инструмент для работы!
Но есть одна проблема, при конвертации Московской базы, скрип постоянно выдает ошибку:
> Fatal error: Out of memory (allocated 1751646208) (tried to allocate 39845867 bytes) in C:\Users\pusto_000\Documents\GitHub\DgdatToXlsx\convert.php on line 273

Скрипт проверил на всех базах 2Гиса, только с Москвой такая проблема. Помогите пожалуйста разобраться, как исправить эту ошибку?

**Comment by emka2k** [comment]
да как так то.
в php.ini поправь сколько памяти выделяешь

**Comment by lagytenko** [comment]
Да, я пробовал по-всякому. По-началу со 128МБ была другая ошибка: 

> Fatal error: Allowed memory size of 134217728 bytes exhausted (tried to allocate 244776319 bytes) in C:\Users\pusto_000\Documents\GitHub\DgdatToXlsx\convert.php on line 86

Cейчас "Out of memory". Тестил на компах с 4-мя и 8-ю ГБ оперативки, выдает одну и ту же ошибку. 

Добавить строку в код:
`ini_set('memory_limit',-1);`
перестало реагировать на измерение memory_limit в php.ini (оставил стандартные 128МБ), но ошибка "Out of memory" осталась.

Может автор может помочь найти решение?

**Comment by emka2k** [comment]
после редактирования сервисы перезапускал ?

**Comment by mbry** [comment]
у меня стоит в php.ini memory_limit = 8000M
На компе 32 Gb

**Comment by mbry** [comment]
php 7

**Comment by lagytenko** [comment]
Установил php 7, убрал лимиты на память `ini_set('memory_limit',-1);`, закоментил выгрузку не нужных мне данных:
```
        //$out[] = iconv('UTF-8','cp1251',$payments);
	//$out[] = iconv('UTF-8','cp1251',$wrk);
	//$out[] = iconv('UTF-8','cp1251',$bld_name);
	//$out[] = iconv('UTF-8','cp1251',$bld_purpose);
	//$out[] = iconv('UTF-8','cp1251',$vk);
	//$out[] = iconv('UTF-8','cp1251',$fb);
	//$out[] = iconv('UTF-8','cp1251',$skype);
	//$out[] = iconv('UTF-8','cp1251',$twitter);
	//$out[] = iconv('UTF-8','cp1251',$insta);
	//$out[] = iconv('UTF-8','cp1251',$icq);
	//$out[] = iconv('UTF-8','cp1251',$jabber); 
```
На других базах файл с выгрузкой заметно полегчал, на Москве выпала та же ошибка, хотя памяти не-хватило уже меньше:

> VirtualAlloc() failed: [0x00000008] Iaainoaoi?ii iaiyoe aey ia?aaioee eiiaiau.
> VirtualAlloc() failed: [0x00000008] Iaainoaoi?ii iaiyoe aey ia?aaioee eiiaiau.
> 
> PHP Fatal error:  Out of memory (allocated 1302331392) (tried to allocate 335585
> 28 bytes) in C:\Users\?????░??\Desktop\DgdatToXlsx\convert.php on line 274
> 
> Fatal error: Out of memory (allocated 1302331392) (tried to allocate 33558528 by
> tes) in C:\Users\?????░??\Desktop\DgdatToXlsx\convert.php on line 274

Заметил баг, первый столбец данных дублируется дважды из-за этого съезжают заголовки.

Все-таки остается не ясно почему не хватает оперативы, и как бы оптимизировать выгрузку, что бы  выгружалась Москва?

**Comment by lagytenko** [comment]
Убрал выгрузку Адресов из декодера и заработало, все-таки компу не хватало ресурсов.
